assets/src/data/model/entity-factory/base-entity.js

```diff
@@ -59,6 +59,7 @@ class BaseEntity {
 		);
 		createEntityGettersAndSetters( this );
 		createPersistingGettersAndSetters( this );
+		Object.seal( this );
```

Comment:
This has the added benefit of making sure there won't be a repeat of the snafu in the store.